docs/DeveloperGuide.md

```diff
@@ -1,34 +1,965 @@
+command the user intends to run and is first word from the user input. The command dictates how `Parser` extracts the 
+parameter and flags. The **parameter** is the argument that comes after the command word and can vary depending on the 
+command. For example, if the command is `add`, the parameter would be the module name, but for `delete`, the parameter 
+would be the module code. Flags come after parameters and are available only for a few commands. They specify the 
+additional information required for the command to run. For `add`, flags would be for module code, module type, MCs and 
+prerequisites.
+
+#### Considerations
+
+From the start, it was known that `Parser` would be one of the more challenging components to implement due to the large
+number of commands and the variance in parameter and flag types. In order to ensure that `Parser` satisfies *Single 
+Responsibility Principle*, the class is implemented in a way that it does no validation check on the correctness or 
+format of the parameter and flags, and is only responsible for validating the number of flags and parameters in user
+input for the given command before passing the parameters and flags to the relevant `Command` class to do further 
+validation of the type and formatting of flags and parameters.
+
+### 4.3 Command
+
+The `command` package is responsible for executing the command from the user. The package contains the  abstract class 
+`Command` and 8 subclasses that inherit `Command`, one for each valid iGraduate command .
+
+#### Details
+
+The abstract class `Command` contains only 1 method: `execute()`, which takes in 3 parameters: `moduleList`, `ui` and
+`storage`. These 3 parameters aid with printing information to the user, making modifications to the data and saving the
+data. Each subclass of `Command` overrides `execute()` and implements their own methods to execute the command. Each 
+subclass also has a unique constructor signature as each subclass requires different parameters to execute.
+
+#### Implementation
+
+The implementation for executing every command differs, and the implementation details of each of them will be further 
+elaborated below.
+
+#### 4.3.1 Add Command
+
+The add command allows a user to add a new module to the list of existing modules. The module name is part of the 
+parameters and is extracted directly from user input while the various information required to add a new module are 
+included in the flags of the user input. There are 3 compulsory flags and 1 optional flag for adding a module:
+
+1. module code
+    - `-c <String>`
+1. module credits
+    - `-mc <double>`
+1. module type
+    - `-t <String>`
+1. (Optional) prerequisite modules
+    - `-p [>String>, ...]`
+
+> ℹ️ **Note:** The order of flags in user input does not matter.
+
+The sequence diagram below shows the execution of add command in action:
+
+![archi](./images/AddCommandSequenceDiagram.png)
```

Comment:
![image](https://user-images.githubusercontent.com/10681776/113076733-45964d00-9202-11eb-90b1-fef3450b969d.png)

For the "cross" you put for AddCommand, would you like to put it lower? It looks like the instance is out of reference even before it returns values for the current diagram.